TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -474,5 +476,162 @@ private SymbolReference CreateTableOrAliasName(CodeElementsParser.TableOrAliasNa
+        {
+            MinValue,
+            MaxValue,
+            Cycle,
+            Order,
+            Cache,
+            Restart,
+            Increment
+        }
+
+        public AlterSequenceStatement CreateAlterSequenceStatement(
+            CodeElementsParser.AlterSequenceStatementContext context)
+        {
+            var duplicates = new List<string>();
+            var clauseSet = new HashSet<AlterSequenceClauseType>();
+            TableViewCorrelationName sequenceName = null;
+            SyntaxProperty<bool> ordered = null;
+            SyntaxProperty<bool> cycle = null;
+            SyntaxProperty<bool> restart = null;
+            SyntaxProperty<bool> hasMaxValue = null;
+            SyntaxProperty<bool> hasMinValue = null;
+            SyntaxProperty<bool> cache = null;
+            SqlConstant minValue = null;
+            SqlConstant maxValue = null;
+            SqlConstant restartValue = null;
+            SqlConstant incrementValue = null;
+            SqlConstant cacheSize = null;
+            if (context.sequence_name != null)
+            {
+                sequenceName = CreateTableOrViewOrCorrelationName(context.sequence_name);
+            }
+
+            foreach (var alterSequenceClauseContext in context.alterSequenceClause())
+            {
+                if (alterSequenceClauseContext.restartClause() != null)
+                {
+                    SetOption(alterSequenceClauseContext.restartClause(), AlterSequenceClauseType.Restart,
+                        c => restart = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(c)));
+                    if (alterSequenceClauseContext.restartClause().numeric_constant != null)
+                    {
+                        restartValue =
+                            new SqlConstant((Token) alterSequenceClauseContext.restartClause().numeric_constant);
+                    }
+                }
+
+                if (alterSequenceClauseContext.incrementClause() != null &&
+                    alterSequenceClauseContext.incrementClause().numeric_constant != null)
+                {
+                    SetOption(alterSequenceClauseContext.incrementClause(), AlterSequenceClauseType.Increment,
+                        c => incrementValue = new SqlConstant((Token) c.numeric_constant));
+                }
+
+                if (alterSequenceClauseContext.minValueClause() != null)
+                {
+                    SetOption(alterSequenceClauseContext.minValueClause(), AlterSequenceClauseType.MinValue,
+                        c => hasMinValue = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(c)));
+                    if (alterSequenceClauseContext.minValueClause().numeric_constant != null)
+                    {
+                        minValue = new SqlConstant((Token) alterSequenceClauseContext.minValueClause()
+                            .numeric_constant);
+                    }
+                }
```

Comment:
For MINVALUE, MAXVALUE and CACHE, I would rather create both properties only if both keyword and value are present.

```csharp
if (alterSequenceClauseContext.minValueClause() != null && alterSequenceClauseContext.minValueClause().numeric_constant != null)
{
    SetOption([...]);
}
```

and in the action provided to SetOption, set both properties (boolean and constant).

Reply:
I didn't change that yet . I have a question .